File: 프로그래머스/3/152995. 인사고과/인사고과.cpp
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int solution(vector<vector<int>> scores) {
    vector<int> wanho = scores[0];
    int wanho_sum = wanho[0] + wanho[1];
    sort(scores.begin(), scores.end(), [](const vector<int> &a, const vector<int> &b){
        if(a[0] != b[0]) return a[0] > b[0];
        return a[1] < b[1];
        });
    int maxs = -1; 
    int rank = 1;
    for(auto s : scores){
        if(s[1] < maxs){
            if(s[0] == wanho[0] && s[1] == wanho[1]) return -1;
            continue;
        }
        maxs = s[1];
        if(s[0] + s[1] > wanho_sum) rank++;
    }
    return rank;
}

int main(){
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    cout << solution({{2,2}, {1,4}, {3,2}, {3,2}, {2,1}});
}

File: 프로그래머스/3/152995. 인사고과/인사고과.py
import sys
def print(*args, sep=" ", end="\n"):
    sys.stdout.write(sep.join(map(str, args)) + end)
def input():
    return sys.stdin.readline().rstrip()

def solution(scores):
    sorted_scores = sorted(scores, key = lambda x: (-x[0], x[1]))
    wanho = tuple(scores[0])
    wanho_sum = sum(wanho)
    maxs = -1
    rank = 1
    for f, s in sorted_scores:
        if s < maxs:
            if (f, s) == wanho:
                return -1
            continue

        maxs = s
        
        if f + s > wanho_sum:
            rank += 1

    return rank

def main():
    print(solution([[2,2],[1,4],[3,2],[3,2],[2,1]]))
if __name__ == '__main__':
    main()

